src/Microsoft.Framework.PackageManager/Restore/RestoreCommand.cs

```diff
@@ -912,12 +909,9 @@ void Display(string indent, IEnumerable<GraphNode> graphs)
 
         private void ReadSettings(string solutionDirectory)
         {
-            Settings = SettingsUtils.ReadSettings(solutionDirectory, NuGetConfigFile, FileSystem, MachineWideSettings);
-
-            // Recreate the source provider and credential provider
-            SourceProvider = PackageSourceBuilder.CreateSourceProvider(Settings);
-            //HttpClient.DefaultCredentialProvider = new SettingsCredentialProvider(new ConsoleCredentialProvider(Console), SourceProvider, Console);
-
+            var config = NuGetConfig.ForSolution(solutionDirectory, FileSystem);
+            Settings = config.Settings;
+            SourceProvider = config.Sources;
```

Comment:
Just trying not to rock the boat too much and leave the code as it was before. I can do that though.